File: Data Structures & Algorithms/word-break/submission-1.cpp
class Solution
{
public:
    bool wordBreak(string s, vector<string> &wordDict)
    {
        vector<bool> dp(s.size() + 1, false);
        dp[s.size()] = true;

        for (int i = s.size() - 1; i >= 0; i--)
        {
            for (string word : wordDict)
            {
                if (dp[i] == true)
                {
                    continue;
                }
                if (i + word.size() <= s.size() && s.substr(i, word.size()) == word)
                {
                    dp[i] = dp[i + word.size()];
                }
            }
        }
        return dp[0];
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        word_map = {}
        for word in strs:
            count = [0] * 26
            for c in word:
                count[ord(c) - ord('a')] += 1
            key = str(count)
            if key in word_map:
                word_map[key].append(word)
            else:
                word_map[key] = [word]
        ans = []
        for key, val in word_map.items():
            ans.append(val)
        return ans